sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2875,19 +2887,14 @@ def get_grpc_channel(self) -> 'grpc.Channel':
                 f'the lock at {lock_id}. '
                 f'{common_utils.format_exception(e)}') from e
 
-    def _cleanup_ssh_tunnel(self, tunnel_info: SSHTunnelInfo) -> None:
-        """Clean up an SSH tunnel by terminating the process."""
+    def _terminate_ssh_tunnel_process(self, tunnel_info: SSHTunnelInfo) -> None:
+        """Terminate the SSH tunnel process."""
         try:
             proc = psutil.Process(tunnel_info.pid)
             if proc.is_running() and proc.status() != psutil.STATUS_ZOMBIE:
                 logger.debug(
                     f'Terminating SSH tunnel process {tunnel_info.pid}')
-                proc.terminate()
-                try:
-                    proc.wait(timeout=3)
-                except psutil.TimeoutExpired:
-                    proc.kill()
-                    proc.wait(timeout=1)
+                subprocess_utils.kill_children_processes(proc)
```

Comment:
We need to kill the children processes too because otherwise, the `kubectl port-forward` process will still exist:
<img width="704" height="276" alt="Image" src="https://github.com/user-attachments/assets/7acb5c79-a9a1-462a-a7d6-d4f1d6a48d7d" />